### PR TITLE
Support navigation nodes that just link to other nodes

### PR DESCRIPTION
**Pull request by jesusreal**
**Description**

Changes proposed in this pull request:

- Support absolute paths in navigation nodes
- Add example in angular application
- Add UI tests
- Add documentation

**Related issue(s)**

### CODE REVIEW

File: docs/navigation-configuration.md [hunk 1]
```diff
@@ -218,6 +224,7 @@ The node parameters are as follows:
 - **pathSegment** specifies the partial URL of the current segment. **pathSegment** must not contain slashes.
   - A static settings example reflects `luigidomain.test/settings`.
   - A dynamic settings example, prefixed with a colon, loads on any other value. 
+- **absolutePath** is a string which refers to an absolute path in the navigation structure. If this parameter is defined, **pathSegment** is ignored.
 - **externalLink** is an object which indicates that the node links to an external URL. If this parameter is defined, **pathSegment** is ignored. It has the following properties:
```
Comment (dariadomagala) [inline docs/navigation-configuration.md]:
```suggestion
 - **externalLink** is an object which indicates that the node links to an external URL. If this parameter is defined, **pathSegment** and ** absolutePath** parameters are ignored. It has the following properties:
```

File: core/examples/luigi-sample-angular/src/assets/extendedConfiguration.js [hunk 2]
```diff
@@ -170,6 +170,11 @@ var projectDetailNavProviderFn = function(context) {
           }
         ]
       },
+      {
+        absolutePath: '/settings',
+        label: 'Go to absolute path',
+        viewUrl: '/sampleapp.html#/settings'
```
Comment (dariadomagala) [inline core/examples/luigi-sample-angular/src/assets/extendedConfiguration.js]: The `viewUrl` is ignored here. This feature redirects to an existing view, doesn't create a new one.

File: docs/navigation-configuration.md [hunk 3]
```diff
@@ -34,6 +34,11 @@ A sample navigation structure looks as follows:
         label: 'Home',
         viewUrl: 'https://my.microfrontend.com/',
         children: [
+          {
+            absolutePath: '/home',
+            label: 'Go back home',
+            viewUrl: 'https://my.microfrontend.com/'
```
Comment (dariadomagala) [inline docs/navigation-configuration.md]: As I wrote in the configuration file, the `viewUrl` is not needed.

Comment (jesusreal) [reply docs/navigation-configuration.md]: Removed

File: docs/navigation-configuration.md [hunk 4]
```diff
@@ -218,7 +223,8 @@ The node parameters are as follows:
 - **pathSegment** specifies the partial URL of the current segment. **pathSegment** must not contain slashes.
   - A static settings example reflects `luigidomain.test/settings`.
   - A dynamic settings example, prefixed with a colon, loads on any other value. 
-- **externalLink** is an object which indicates that the node links to an external URL. If this parameter is defined, **pathSegment** is ignored. It has the following properties:
+- **absolutePath** is a string which refers to an absolute path in the navigation structure. If this parameter is defined, **pathSegment** is ignored.
+ - **externalLink** is an object which indicates that the node links to an external URL. If this parameter is defined, **pathSegment** and ** absolutePath** parameters are ignored. It has the following properties:
```
Comment (dariadomagala) [inline docs/navigation-configuration.md]:
```suggestion
 - **externalLink** is an object which indicates that the node links to an external URL. If this parameter is defined, **pathSegment** and **absolutePath** parameters are ignored. It has the following properties:
```

File: docs/navigation-configuration.md [hunk 5]
```diff
@@ -20,8 +20,8 @@ The navigation structure is a recursive tree-like data structure that defines al
 
 A navigation path is any existing path in the navigation tree. It connects the following elements together:
 
-- The **pathSegment** of the main application. The **pathSegment** is a part of the URL in the browser URL bar.
-- The **viewUrl** of a micro front-end rendered in the content area of the application.
+- The path of the main application, i.e, the path in the browser URL. The path is defined in a Luigi navigation node through one of these three parameters, listed in order of precedence: **externalLink**, **absolutePath**, and **pathSegment**.
```
[the remark this answers is not in the document]
Comment (jesusreal) [reply docs/navigation-configuration.md]: done

File: docs/navigation-configuration.md [hunk 6]
```diff
@@ -218,7 +227,8 @@ The node parameters are as follows:
 - **pathSegment** specifies the partial URL of the current segment. **pathSegment** must not contain slashes.
   - A static settings example reflects `luigidomain.test/settings`.
   - A dynamic settings example, prefixed with a colon, loads on any other value. 
-- **externalLink** is an object which indicates that the node links to an external URL. If this parameter is defined, **pathSegment** is ignored. It has the following properties:
+- **link** is a string which refers to an absolute path in the navigation structure or a relative path to a grand child of the current path. If this parameter is defined, **pathSegment** is ignored.
```
Comment (dariadomagala) [inline docs/navigation-configuration.md]: Why does this child has to be grand? ;)

Comment (dariadomagala) [reply docs/navigation-configuration.md]: Makes sense. It's ok, just change it from `grand child` to `grandchild` ;)

**Pull request merged**